File: Results/DER/Translation/translation_output/no_test/magicoder/codenet/python/java-sanitized/p02775.java
import java.util.Scanner;

public class p02775 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.next();
        int[] n = new int[input.length() + 1];
        for (int i = 0; i < input.length(); i++) {
            n[i] = input.charAt(input.length() - 1 - i) - '0';
        }
        n[input.length()] = 0;
        int s = 0;
        int res = 0;
        for (int i = 0; i < n.length - 1; i++) {
            int k = n[i] + s;
            if (k < 5 || (k == 5 && n[i + 1] < 5)) {
                res += k;
                s = 0;
            } else {
                res += 10 - k;
                s = 1;
            }
        }
        res += s;
        System.out.println(res);
    }
}

File: Results/DER/Translation/translation_output/misleading_test/deepseekcoder/codenet/java/python-sanitized/p02775.py
def p02775():
    N = input()
    l = len(N) + 1

    # 0 ãã¤ã, 1 ã´ã£ãã, 2 å¤ã
    dp = [[0 for _ in range(3)] for _ in range(l)]

    dp[0][0] = dp[0][2] = 1

    for i in range(1, l - 1):
        c = int(N[i])
        dp[i][0] = min(dp[i - 1][0], dp[i - 1][2]) + 9 - c
        dp[i][1] = min(dp[i - 1][0] + 1, dp[i - 1][1]) + c
        dp[i][2] = min(dp[i - 1][0] + 1, dp[i - 1][1]) + c + 1

    dp[l - 2][0] += 1
    print(min(dp[l - 2][0], min(dp[l - 2][1], dp[l - 2][2])))

p02775()